src/main/scala/com/fulcrumgenomics/bam/ZipperBams.scala

```diff
@@ -240,6 +240,15 @@ class ZipperBams
       }
     }
 
+    // There really should be no more mapped reads!
+    if (mappedIter.hasNext) {
+      throw new IllegalStateException(
+        """Error: processed all unmapped reads but there are mapped reads remaining to be read.
+        |Please ensure the unmapped and mapped reads have the same set of read names in the same
+        |order, and reads with the same name are consecutive (grouped) in each input""".stripMargin
+      )
+    }
+
     out.close()
```

Comment:
It would probably be kind to move `out.close()` above so that the output BAM is at least somewhat valid/intact?